spatialmath/box.go

```diff
@@ -352,3 +355,91 @@ func separatingAxisTest(positionDelta, plane r3.Vector, halfSizeA, halfSizeB [3]
 	}
 	return sum
 }
+
+// ToPointCloud converts a box geometry into a []r3.Vector.
+func (b *box) ToPointCloud(resolution float64) []r3.Vector {
+	// check for user defined spacing
+	var iter float64
+	if resolution != 0. {
+		iter = resolution
+	} else {
+		iter = defaultBoxPointDensity
+	}
+
+	// create points on box faces with box centered at (0, 0, 0)
+	// TODO: add more detailed explanation
+	var faces []r3.Vector
+	fillFaces := func(fixedDimension int, check1, check2, check3 string) {
+		starts := [3]float64{0.0, 0.0, 0.0}
+		starts[fixedDimension] = b.halfSize[fixedDimension]
+		for i := starts[0]; compares(check1, i, b.halfSize[0]); i += iter {
+			for j := starts[1]; compares(check2, j, b.halfSize[1]); j += iter {
+				for k := starts[2]; compares(check3, k, b.halfSize[2]); k += iter {
+					p1 := r3.Vector{i, j, k}
+					p2 := r3.Vector{i, j, -k}
+					p3 := r3.Vector{i, -j, k}
+					p4 := r3.Vector{i, -j, -k}
+					p5 := r3.Vector{-i, j, k}
+					p6 := r3.Vector{-i, j, -k}
+					p7 := r3.Vector{-i, -j, -k}
+					p8 := r3.Vector{-i, -j, k}
+
+					switch {
+					case i == 0.0 && j == 0.0:
+						faces = append(faces, p1, p2)
+					case j == 0.0 && k == 0.0:
+						faces = append(faces, p1, p5)
+					case i == 0.0 && k == 0.0: // edited now
+						faces = append(faces, p1, p7)
+					case i == 0.0:
+						faces = append(faces, p1, p2, p3, p4)
+					case j == 0.0:
+						faces = append(faces, p1, p2, p5, p6)
+					case k == 0.0:
+						faces = append(faces, p1, p3, p5, p8)
+					case i != 0.0 && j != 0.0 && k != 0.0:
+						faces = append(faces, p1, p2, p3, p4, p5, p6, p7, p8)
+					}
+				}
+			}
+		}
+	}
+
+	// TODO: the for loop below can be made concurrent if the ToPointCloud method is too slow
+	for i := 0; i < 3; i++ {
+		// TODO: add more detailed explanation
```

Comment:
yeah adding some comments describing how these comparison signs ignore common edges (I think that is what they are doing?) would be good